tjreports/administrator/controllers/tjreport.php

```diff
@@ -36,8 +36,19 @@ public function __construct()
 	 */
 	public function getplugins()
 	{
+		$app     = JFactory::getApplication();
```

Comment:
Can not we create tjreport.json.php can move this functions there? You will need to pass format=json to your ajax urls.